### ISSUE TITLE
Regarding the Loss Function

### ISSUE THREAD

**Issue by jingfengou** [yulun-rayn/graphVCI#2]
Hello, after reading your paper and code, I found that the loss function in your paper does not match the one in your code. Could you please explain how the reconstruction loss for Y and the distribution loss for Y' are implemented in the paper? Thank you!

**Comment by yulun-rayn** [comment]
Hi, thanks for the question. Which part of the loss function does not match the paper in your opinion?

Here is the reconstruction loss: https://github.com/yulun-rayn/variational-causal-inference/blob/39aab90d3ec8e6101312648c66b71f028b45cecb/vci/model/model.py#L465

Here is the distribution loss: https://github.com/yulun-rayn/variational-causal-inference/blob/39aab90d3ec8e6101312648c66b71f028b45cecb/vci/model/model.py#L470

**Comment by jingfengou** [comment]
> Hi, thanks for the question. Which part of the loss function does not match the paper in your opinion?
> 
> Here is the reconstruction loss: https://github.com/yulun-rayn/variational-causal-inference/blob/39aab90d3ec8e6101312648c66b71f028b45cecb/vci/model/model.py#L465
> 
> Here is the distribution loss: https://github.com/yulun-rayn/variational-causal-inference/blob/39aab90d3ec8e6101312648c66b71f028b45cecb/vci/model/model.py#L470

Yes, you're right. The reconstruction loss you mentioned typically refers to the Mean Squared Error (MSE) between the predicted outputs (Y predicted) and the actual labels (Y actual) in my understanding. But the code or context you're referring to looks like it's involving a kind of distribution loss, right？Additionally, I'd like to ask if the T and Y (T' and Y') in your dataset correspond one-to-one? Is there a one-to-many situation where one experiment involves multiple cell expressions? If so, would this affect the training of the model? Or is the distribution loss you're using intended to avoid directly calculating the MSE before and after perturbation?

**Comment by yulun-rayn** [comment]
Reconstruction loss is not actually MSE. It is only MSE if you assume the outcome distrubution to be Gaussian and assume the variance of outcome distribution to be constant, in which case reconstruction loss reduces to MSE.

As for one-to-many, the simplest way is just to have many output heads in the decoder to simultaneous predict each outcome, i.e. essentially concatenating many outcomes to one. But this naive approach might not be the best way to handle specific multimodal data and if you have better idea, I encourage you to try it out and it could be publication worthy :)

**Comment by jingfengou** [comment]
Thanks very much！

**Comment by jingfengou** [comment]
In fact, I would like to know how to handle the situation where there will be different expression data under the same disturbance?
